pkg/sql/compile/compile.go

```diff
@@ -344,6 +344,9 @@ func (c *Compile) compilePlanScope(n *plan.Node, ns []*plan.Node) ([]*Scope, err
 	case plan.Node_TABLE_SCAN:
 		ss := c.compileTableScan(n)
 		return c.compileSort(n, c.compileProjection(n, c.compileRestrict(n, ss))), nil
+	case plan.Node_EXTERNAL_SCAN:
+		ss := c.compileExternalScan(n)
+		return c.compileSort(n, c.compileProjection(n, c.compileRestrict(n, ss))), nil
```

Comment:
Another surprise, what is compileSort?   Why do we need sort?

Reply:
External scan There will be no filter, projection, sorting or limit information at all, so why write it that way?